TransferList/CustomDragImage.js

```diff
@@ -0,0 +1,86 @@
+import React, {useCallback, useState} from 'react';
+import {flushSync} from 'react-dom';
+
+import css from './TransferList.module.less';
+import itemCss from '../Item/Item.module.less';
+
+const DragImage = React.forwardRef((props, ref) => {
+	const getCommonElementStyles = useCallback (() => {
+		const item = document.querySelectorAll(`.${css.draggableItem}`)[0];
+		if (!item) {
+			return {};
+		}
+
+		const baseStyles = {
+			left: '0px',
+			border: '1px solid black',
+			position: "absolute",
+			height: item.clientHeight + 'px',
+			width: item.clientWidth + 'px'
+		};
+
+		if (props.listComponent === 'VirtualList') {
+			const itemBg = item.querySelectorAll(`.${itemCss.bg}`)[0];
+
+			return {
+				...baseStyles,
+				backgroundColor: window.getComputedStyle(itemBg).backgroundColor,
+				borderRadius: window.getComputedStyle(itemBg).borderRadius
+			};
+		}
+
+		return {
+			...baseStyles,
+			backgroundColor: window.getComputedStyle(item, ':before').backgroundColor,
+			borderRadius: window.getComputedStyle(item, ':before').borderRadius
+		};
+	}, [props.listComponent]);
+
+	const generateMultiDragImage = useCallback(() => {
+		const item = document.querySelectorAll(`.${css.draggableItem}`)[0];
+		if (!item) {
+			return;
+		}
+
+		return <div style={{...getCommonElementStyles(), height: 1.6 * item.clientHeight + 'px', top: 0, zIndex: '-110'}}>
+			<div style={{...getCommonElementStyles(), top: 0, zIndex: '-100'}} />
+			<div style={{...getCommonElementStyles(), top: 0.3 * item.clientHeight + 'px', zIndex: '-90'}} />
+			<div style={{...getCommonElementStyles(), top: 0.6 * item.clientHeight + 'px', zIndex: '-80'}} />
+		</div>;
+	}, [getCommonElementStyles]);
+
+	let [content, setContent] = useState(null);
+	let domRef = React.useRef(null);
+
+	React.useImperativeHandle(ref, () => (isSingle, callback) => {
```

Comment:
you don;t need to use "React." just import useImperativeHandle from 'react" and use it as is